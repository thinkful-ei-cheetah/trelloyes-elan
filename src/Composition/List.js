import React from 'react';
import Card from './Card';
import './List.css';

// implemented destructuring {props} ==> {header, cards}

function List({header, cards}) {
    const cardsArray = cards.map((card, i) => <Card title={card.title} content={card.content} key={i} />);
    return (
    <div>
        <section className="list">
            <header>{header}</header>
            <div className="List-cards">
                {cardsArray}
            </div>
        </section>
    </div>
    )
}

export default List;