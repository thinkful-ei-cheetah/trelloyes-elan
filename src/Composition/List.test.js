import React from 'react';
import ReactDOM from "react-dom";
import Card from './Card';
import STORE from '../store';
import List from './List';
import renderer from 'react-test-renderer';

describe('List component', () => {
    // SMOKE
    it('renders', () => {
      const div = document.createElement('div');
      ReactDOM.render(<List header="test" cards={STORE.lists.map((card, i) => <Card title={card.title} content={card.content} key={i} />)} />, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    // SNAPSHOT
    it('renders the UI as expected', () => {
      const tree = renderer
        .create(<List header="test" cards={STORE.lists.map((card, i) => <Card title={card.title} content={card.content} key={i} />)} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });