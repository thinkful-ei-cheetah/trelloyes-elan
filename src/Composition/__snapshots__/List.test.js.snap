// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<div>
  <section
    className="list"
  >
    <header>
      test
    </header>
    <div
      className="List-cards"
    >
      <div
        className="Card"
      >
        <h3 />
        <p />
      </div>
      <div
        className="Card"
      >
        <h3 />
        <p />
      </div>
      <div
        className="Card"
      >
        <h3 />
        <p />
      </div>
      <div
        className="Card"
      >
        <h3 />
        <p />
      </div>
    </div>
  </section>
</div>
`;
